python/vowpalwabbit/DFtoVWtoTensorboard.py

```diff
@@ -0,0 +1,208 @@
+from vowpalwabbit import pyvw
+from datetime import datetime
+import tensorboardX as tx
+
+
+class DFtoVWtoTensorboard:
+	"""This class provides capability for learning possible on DFtoVW object, provides capability to 
+	print metrics (average_loss, since_last, label, prediction, num_features) and also provides support 
+	to log these metrics for Tensorboard Visualization"""
+
+	def __init__(self, df_to_vw, vw):
+		"""Construct a DFtoVWtoTensorboard object
+
+		Parameters
+		----------
+
+		df_to_vw : A DFtoVW object
+					This would be used to construct examples 
+		vw		 : A vowpalwabbit object
+ 					This would be used for learning using the algorithm specified in the object, and also to compute a few metrics
+
+		Returns
+		-------
+
+		self : DFtoVWtoTensorboard
+		"""
+		self.df_to_vw = df_to_vw
+		self.vw = vw
+		self.vw_formatted_data = df_to_vw.convert_df()
```

Comment:
The dependency on `DFtoVW` is not necessary in my opinion.

Indeed, the list of examples need not be the result of the  `DFtoVW.convert_df` method. The user could just provide the list of examples instead.

Reply:
So should the user providing the list of examples be **optional behaviour**? or `DFtoVW ` object passed as `df_to_vw` should not be used at all to get the list of example and the user should provide the list of examples always?

Reply:
The way you can do this is similar to how the `pyvw.learn` method deals with different types of input: Check the type explicitly.